### PR TITLE
Add Google Analytics reporting in place of existing custom reporting

### PR DESCRIPTION
**Pull request by sanketsudake**

### CODE REVIEW

File: charts/fission-all/templates/post-install-job.yaml [hunk 1]
```diff
@@ -22,15 +22,20 @@ spec:
       labels:
         release: {{ .Release.Name }}
         app: {{ template "name" . }}
-      annotations:
-        {{- if .Values.enableIstio }}
-        "sidecar.istio.io/inject": "false"
-        {{- end }}
```
Comment (vishal-biyani) [inline charts/fission-all/templates/post-install-job.yaml:28]: What is the intention of removing this one?

File: charts/fission-all/templates/post-upgrade-job.yaml [hunk 2]
```diff
@@ -22,15 +22,20 @@ spec:
       labels:
         release: {{ .Release.Name }}
         app: {{ template "name" . }}
-      annotations:
-        {{- if .Values.enableIstio }}
-        "sidecar.istio.io/inject": "false"
-        {{- end }}
```
Comment (vishal-biyani) [inline charts/fission-all/templates/post-upgrade-job.yaml:28]: Similar to previous Istio annotation comment

File: charts/fission-core/templates/nonhelm-install-job.yaml [hunk 3]
```diff
@@ -26,6 +26,16 @@ spec:
...
-          image: "fission/alpinecurl"
-          command: ["sh", "-c", "/usr/bin/curl -m 5 -H \"Content-Type: application/json\" -X POST -d '{\"type\": \"yaml-post-install\", \"chartName\": \"{{ .Chart.Name }}\", \"chartVersion\": \"{{ .Chart.Version }}\"}' https://g.fission.io/analytics || true"]
+        {{- if .Values.imageTag }}
+          image: {{ .Values.postInstallReportImage }}:{{ .Values.imageTag }}
+        {{- else }}
+          image: {{ .Values.postInstallReportImage }}
+        {{- end }}
+          imagePullPolicy: {{ .Values.pullPolicy }}
+          command: [ "/reporter" ]
+          args: ["event", "-c", "fission-use", "-a", "helm-post-install", "-l", "{{ .Chart.Name }}-{{ .Chart.Version }}"]
```
Comment (vishal-biyani) [inline charts/fission-core/templates/nonhelm-install-job.yaml:36]: If this file is used for non helm install - then should the `helm-post-install` not be `yaml-post-install` or something similar?

File: hack/release-build.sh [hunk 4]
```diff
@@ -85,6 +85,19 @@ build_builder_image() {
     docker tag $tag fission/builder:latest
 }
 
+# Build pre-upgrade-checks image
```
Comment (vishal-biyani) [inline hack/release-build.sh:88]: I think misplaced comment

File: test/test_utils.sh [hunk 5]
```diff
@@ -96,6 +97,19 @@ build_and_push_pre_upgrade_check_image() {
...
+    image_tag=$1
+    cache_image=$2
+    travis_fold_start build_and_push_reporter_image $image_tag
+
+    docker build -q -t $image_tag -f $ROOT/cmd/reporter/Dockerfile.reporter --cache-from ${cache_image} --build-arg GITCOMMIT=$(getGitCommit) --build-arg BUILDDATE=$(getDate) --build-arg BUILDVERSION=$(getVersion) .
+
+    gcloud_login
+
+    gcloud docker -- push $image_tag &
+    travis_fold_end build_and_push_reporter_image
```
Comment (vishal-biyani) [inline test/test_utils.sh:110]: 1. We are not using the GCR and Travis both
2. I think we should not doing analytics part as part of test - that will skew results.

File: test/build.sh [hunk 6]
```diff
@@ -24,6 +24,7 @@ setupCIBuildEnv
 build_and_push_go_mod_cache_image $REPO/go-mod-image-cache
 build_and_push_fission_bundle $REPO/$IMAGE:$TAG $REPO/go-mod-image-cache
 build_and_push_pre_upgrade_check_image $PRE_UPGRADE_CHECK_IMAGE:$TAG $REPO/go-mod-image-cache
+build_and_push_reporter_image $REPORTER_IMAGE:$TAG $REPO/go-mod-image-cache
```
Comment (vishal-biyani) [inline test/build.sh:27]: I think not needed as part of build every time for test. Will increase build time too

File: skaffold.yaml [hunk 7]
```diff
@@ -41,6 +44,7 @@ deploy:
         artifactOverrides:
           image: "fission"
           preUpgradeChecksImage: "preupgradechecks"
+          postInstallReportImage: "reporter"
```
Comment (vishal-biyani) [inline skaffold.yaml:47]: Skaffold should turn analytics off.

**Review by vishal-biyani — CHANGES_REQUESTED** [review]
Some very minor comments, LGTM otherwise

File: charts/fission-all/templates/post-install-job.yaml (hunk 1 above)
Comment (sanketsudake) [reply charts/fission-all/templates/post-install-job.yaml:28]: Added back, as I don't see any value of Istio injection in these post hooks.

**Comment by sanketsudake** [comment]
@RealHarshThakur Could you please elaborate more on it ? Probably I don't have the context to the solution you are referring to. Any references you can point to from fission-bundle?

**Comment by RealHarshThakur** [comment]
@sanketsudake  I mean we run a lot of components from Fission-Bundle itself(Router, Executor, MQT-KEDA) , so I was thinking if it made sense to keep reporter as part of it too.  https://github.com/fission/fission/blob/master/cmd/fission-bundle/main.go

**Pull request merged**